projects/RabbitMQ.Client/client/impl/WireFormatting.cs

```diff
@@ -162,15 +168,22 @@ public static unsafe int ReadShortstr(ReadOnlySpan<byte> span, out string value)
             // equals span.Length >= byteCount + 1
             if (span.Length > byteCount)
             {
-                fixed (byte* bytes = &span.Slice(1).GetPinnableReference())
+#if NETCOREAPP
+                value = Encoding.UTF8.GetString(span.Slice(1, byteCount));
+                return 1 + byteCount;
+#else
+                unsafe
                 {
-                    value = Encoding.UTF8.GetString(bytes, byteCount);
-                    return 1 + byteCount;
+                    fixed (byte* bytes = &MemoryMarshal.GetReference(span.Slice(1)))
+                    {
+                        value = Encoding.UTF8.GetString(bytes, byteCount);
+                        return 1 + byteCount;
+                    }
                 }
+#endif
             }
 
-            value = string.Empty;
-            return ThrowArgumentOutOfRangeException(span.Length, byteCount + 1);
+            throw new ArgumentOutOfRangeException("span", $"Span has not enough space ({span.Length} instead of {byteCount + 1})");
```

Comment:
Most of the methods in this class are inlined. The throw methods helped to reduce the imported code size and thus improved jit time and potentially performance itself.
Was there a specific reason to change these back? 
(You'll see the difference if you add `[DisassemblyDiagnoser]` to the benchmarks

Reply:
Hadn't thought about that. Just looked like it added extra instructions, but didn't look at the disassembly. It's more readable this way though i.m.o.

Reply:
Yes I agree it's less readable, this is also why the pattern isn't used everywhere. Here though I think it will be overall worth the small readability penalty.